### ISSUE TITLE
periodicity of exponential function

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#18747]
from #18736 

> [periodicity] fails with [exponential functions with] ...real and imaginary parts, although in this case period() does work.
```python
from sympy import *
x = Symbol('x', real=True)

e0 = exp(S.ImaginaryUnit*S.Pi*x/4)
print(periodicity(e0, x), '-> OK')

e1 = exp(S.ImaginaryUnit*S.Pi*x/4 + S(10))
print(periodicity(e1, x), '-> OK')

e2 = exp(S.ImaginaryUnit*S.Pi*x/4 + S.Pi/4)
print(periodicity(e2, x), '-> OK')

e3 = exp(S.ImaginaryUnit*S.Pi*(x/4 + S(1)/4))
print(periodicity(e3, x), '-> Not OK')
print(periodicity(re(e3), x), '-> Not OK')
print(periodicity(im(e3), x), '-> Not OK')
print(re(e3).period(), '-> OK')
print(im(e3).period(), '-> OK')

e4 = exp(expand_mul(S.ImaginaryUnit*S.Pi*(x/4 + S(1)/4)))
print(periodicity(e4, x), '-> Not OK')
print(periodicity(re(e4), x), '-> Not OK')
print(periodicity(im(e4), x), '-> Not OK')
print(re(e4).period(), '-> OK')
print(im(e4).period(), '-> OK')
```

**Comment by smichr** [comment]
The code
https://github.com/sympy/sympy/blob/master/sympy/calculus/util.py#L483-L488 may need to handle the re and im parts individually rather than failing when there is an im part.

**Comment by j3parera** [comment]
Perhaps the solution involves creating the `period` method in the `exp` function. This method should check if the exponent matches the pattern i(&omega;x + &phi;). If it matches, the exponential could be expanded as: cos(&omega;x+&phi;) + isin(&omega;x + &phi;) and its period is the same in the real and imaginary parts: 2*&pi;/&omega;. In any other case the exponential is not periodic.

With regard to the errors `periodicity(re(e3), x)` and `periodicity(im(e3), x)` I think that the fix of #18736 will also fix this because `periodicity` will call the fixed code.

**Comment by smichr** [comment]
The recent commit fixed this by handling `exp` in `periodicity`. Please open again if you find other issues.
